src/inputs/AddressInput/index.tsx

```diff
@@ -24,95 +31,137 @@ type AddressInputProps = {
   disabled?: boolean;
   onChangeAddress: (address: string) => void;
   getAddressFromDomain?: (name: string) => Promise<string>;
+  customENSThrottleDelay?: number;
   showLoadingSpinner?: boolean;
 } & TextFieldInputProps;
 
 function AddressInput({
   name,
   address,
   networkPrefix,
-  showNetworkPrefix,
+  showNetworkPrefix = true,
   disabled,
   onChangeAddress,
   getAddressFromDomain,
+  customENSThrottleDelay,
   showLoadingSpinner,
   InputProps,
+  inputProps,
+  hiddenLabel = true,
   ...rest
 }: AddressInputProps): ReactElement {
-  const [prefix, setPrefix] = useState(networkPrefix);
   const [isLoadingENSResolution, setIsLoadingENSResolution] = useState(false);
-
+  const defaulInputValue = addPrefix(address, networkPrefix, showNetworkPrefix);
+  const inputRef = useRef({ value: defaulInputValue });
```

Comment:
Perhaps performance could be improved by following an approach as described [here](https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily) and defining a fn for lazy initializing the ref and then calling it for getting the reference value. But only suggestion because addPrefix guess is not very expensive

Reply:
I think in this case we can omit this, to make the component clearer.